File: go/rna-transcription/rna_transcription.go
// Package strand provides utility to convert DNA strand to its complementary RNA
package strand

import (
	"fmt"
	"strings"
)

// ToRNA returns the RNA complement of a given DNA sequence.
func ToRNA(dna string) string {
	sb := strings.Builder{}

	for i, nucleotide := range dna {
		switch nucleotide {
		case 'G':
			sb.WriteRune('C')
		case 'C':
			sb.WriteRune('G')
		case 'T':
			sb.WriteRune('A')
		case 'A':
			sb.WriteRune('U')
		default:
			return fmt.Sprintf("%c at the position %d is not a DNA nucleotide", nucleotide, i)
		}
	}

	return sb.String()
}

File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(defn to-rna
  "Returns the RNA complement of the given DNA string sequence."
  [dna]
  (let [rna (map {\G \C \C \G \T \A \A \U} dna)]
    (when-not (some nil? rna) (apply str rna))))